server/user_test.go

```diff
@@ -54,7 +54,7 @@ func TestRouteUserStart(t *testing.T) {
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
 			request := httptest.NewRequest("GET", routeUserStart, nil)
-			request.Header.Set("Mattermost-User-Id", tc.userID)
+			request.Header.Set("Mattermost-User-ID", tc.userID)
```

Comment:
That is a predefined header send by the Mattermost Server. It can and should stay as `Mattermost-User-Id`. Otherwise authentication breaks.